### PR TITLE
Remove unused legacy VSIX and set IsProductComponent=true

### PR DESCRIPTION
**Pull request by DefaultRyan**
Lights up the IsProductComponent variable for Component builds that are destined for VS. This option streamlines the insertion process.

The legacy combined VSIX is no longer used and removing it will save us some build time.

### CODE REVIEW

File: dev/VSIX/Extension/Cpp/Dev17/WindowsAppSDK.Cpp.Extension.Dev17.csproj [hunk 1]
```diff
@@ -10,6 +10,8 @@
     <BaseAssemblyName>WindowsAppSDK.Cpp.Extension.Dev17</BaseAssemblyName>
     <AssemblyName Condition="'$(EnableExperimentalVSIXFeatures)'!='true'">$(BaseAssemblyName)</AssemblyName>
     <AssemblyName Condition="'$(EnableExperimentalVSIXFeatures)'=='true'">$(BaseAssemblyName).Experimental</AssemblyName>
+	<JsonName Condition="'$(EnableExperimentalVSIXFeatures)'!='true'">Microsoft.WindowsAppSDK.Cpp.Dev17.json</JsonName>
```
Comment (Scottj1s) [inline dev/VSIX/Extension/Cpp/Dev17/WindowsAppSDK.Cpp.Extension.Dev17.csproj:13]: nit: tabs

**Review by Scottj1s — APPROVED** [review]

**Review by evelynwu-msft — APPROVED** [review]
:shipit:

File: dev/VSIX/Extension/Cpp/Dev17/WindowsAppSDK.Cpp.Extension.Dev17.csproj (hunk 1 above)
Comment (DefaultRyan) [reply dev/VSIX/Extension/Cpp/Dev17/WindowsAppSDK.Cpp.Extension.Dev17.csproj:13]: Nice catch, thanks!

**Pull request merged**